plugins/modules/netconf_config.py

```diff
@@ -27,10 +27,13 @@
   content:
     description:
     - The configuration data as defined by the device's data models, the value can
-      be either in xml string format or text format. The format of the configuration
-      should be supported by remote Netconf server. If the value of C(content) option
-      is in I(xml) format in that case the xml value should have I(config) as root tag.
-    type: str
+      be either in xml string format or text format or json string format.
```

Comment:
"json string format" seems a little unclear to me. It looks like you're expecting to also support a dictionary representation of the JSON (hence why the type change to raw), but nothing in the description would lead me to expect that would be true.